tuna/example/example_lib.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+###############################################################################
+#
+# MIT License
+#
+# Copyright (c) 2023 Advanced Micro Devices, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+"""Example library, integrated with MITuna, runs 'rocminfo' cmd"""
+
+import sys
+
+from tuna.mituna_interface import MITunaInterface
+from tuna.parse_args import TunaArgs, setup_arg_parser, clean_args, args_check
+from tuna.utils.miopen_utility import load_machines
+from tuna.libraries import Library
+from tuna.utils.db_utility import create_tables
+from tuna.example.example_tables import get_tables
+from tuna.example.example_worker import ExampleWorker
+from tuna.example.session import SessionExample
+
+
+class Example(MITunaInterface):
+  """Class to support an example of 'romcinfo' run"""
+
+  def __init__(self):
+    super().__init__(library=Library.EXAMPLE)
+    self.args = None
+
+  def parse_args(self):
+    # pylint: disable=too-many-statements
+    """Function to parse arguments"""
+    parser = setup_arg_parser('Example library integrated with MITuna', [
+        TunaArgs.ARCH, TunaArgs.NUM_CU, TunaArgs.VERSION, TunaArgs.SESSION_ID,
+        TunaArgs.MACHINES, TunaArgs.REMOTE_MACHINE, TunaArgs.LABEL,
+        TunaArgs.RESTART_MACHINE, TunaArgs.DOCKER_NAME
+    ])
+
+    parser.add_argument('--init_session',
+                        action='store_true',
+                        dest='init_session',
+                        help='Set up a new tuning session.')
+
+    clean_args('EXAMPLE', 'example')
+    args = parser.parse_args()
+    if len(sys.argv) == 1:
+      parser.print_help()
+      sys.exit(-1)
+
+    args_check(args, parser)
+
+    return args
+
+  def launch_worker(self, gpu_idx, f_vals, worker_lst, args):
```

Comment:
Logging or error messages can help for this function...Usually with launching a session/worker is where things tend to break and perhaps detailed breakdown - with logging - on what failed, it can help with debugging or configuration for the user endpoint

Reply:
the worker class has logging, so all the function calls in this function log info